### DISCUSSION TITLE
Match should allow changing passwords when using the s3 storage mode

### DISCUSSION THREAD

**Discussion by ryechus** [fastlane/fastlane#17420] [Ideas]
Today I went to change the password for my match store and got this message. This seems like a requirement for the functionality regardless of storage method.

```
[!] Only git-based match allows you to change your password, current `storage_mode` is s3
```

**Comment by rogerluan** [comment]
Hey @ryechus 👋 

There's a comment around that message:

```ruby
# Only git supports changing the password
# All other storage options will most likely use more advanced
# ways to encrypt files
```

Looks like this was implemented like this intentionally for some reason. @joshdholtz do you know if there's a good reasoning behind this limitation, or is it just a pending feature? https://github.com/fastlane/fastlane/pull/13589

**Reply by joshdholtz** [reply]
Huh.... I do not actually know if there is a good reasoning behind that! My guess is that the second option for storage was Google Cloud which has encryption build in by default. S3 was added by another contributor so this may not have been thought about or missed 🤷‍♂️

**Comment by ryechus** [comment]
Hey @joshdholtz, I just want to bump this again because I see that GCS is supported now. Is there a specific limitation around building this feature? Maybe I can offer some assistance.

**Reply by joshdholtz** [reply]
No limitation that I can think of! If somebody would like to take a stab at this that would be 🔥 I’d be happy to review and test 😊 

I have too much lined up right now that I wouldn’t be able to get to this right way 😔

**Comment by AustinFuller** [comment]
So I'm currently facing a rather confusing issue on GitHub Actions related to the match password. I successfully download everything from s3 and then it fails to decrypt. The thing is that to my understanding match passwords aren't relevant when using s3?

**Reply by TapTap21** [reply]
I also need help with this
